internal/provider/resource_docker_container_funcs.go

```diff
@@ -824,39 +824,47 @@ func resourceDockerContainerUpdate(ctx context.Context, d *schema.ResourceData,
 func resourceDockerContainerDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ProviderConfig).DockerClient
 
-	if d.Get("rm").(bool) {
```

Comment:
I had no clue TBH why this was there, as it will never stop/remove the container if `rm=true` 🤔